src/layer/arm/hardsigmoid_arm.cpp

```diff
@@ -0,0 +1,73 @@
+// Tencent is pleased to support the open source community by making ncnn available.
+//
+// Copyright (C) 2017 THL A29 Limited, a Tencent company. All rights reserved.
+//
+// Licensed under the BSD 3-Clause License (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// https://opensource.org/licenses/BSD-3-Clause
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include "hardsigmoid_arm.h"
+
+#if __ARM_NEON
+#include <arm_neon.h>
+#endif // __ARM_NEON
+
+namespace ncnn {
+
+DEFINE_LAYER_CREATOR(HardSigmoid_arm)
+
+int HardSigmoid_arm::forward_inplace(Mat& bottom_top_blob, const Option& opt) const
+{
+    int w = bottom_top_blob.w;
+    int h = bottom_top_blob.h;
+    int channels = bottom_top_blob.c;
+    int size = w * h;
+
+    #pragma omp parallel for num_threads(opt.num_threads)
+    for (int q=0; q<channels; q++)
+    {
+        float* ptr = bottom_top_blob.channel(q);
+
+#if __ARM_NEON
+        int nn = size >> 2;
+        int remain = size - (nn << 2);
+#else
+        int remain = size;
+#endif // __ARM_NEON
+
+#if __ARM_NEON
+        float32x4_t _zero = vdupq_n_f32(0.f);
+        float32x4_t _one = vdupq_n_f32(1.f);
+        while (nn--)
+        {
+            float32x4_t _p = vld1q_f32(ptr);
+            float32x4_t _ans = vdupq_n_f32(0.5f);
+            _ans = vmlaq_n_f32(_ans, _p, 0.2f);
+            _ans = vmaxq_f32(_ans, _zero);
+            _ans = vminq_f32(_ans, _one);
+            vst1q_f32(ptr, _ans);
+
+            ptr += 4;
+        }
+#endif // __ARM_NEON
+        for (int i=0; i<remain; i++)
```

Comment:
tail loop index is wrong
you can refer https://github.com/Tencent/ncnn/blob/master/src/layer/arm/sigmoid_arm.cpp to learn how to handle the main vectorized loop and remaining data properly

Reply:
Not certain about this. Is the currect implementation efficient?